src/Psl/File/temporary.php

```diff
@@ -15,9 +15,13 @@
  *                                         the system default temporary directory.
  * @param non-empty-string|null $prefix The prefix of the generated temporary filename.
  *
+ * @throws Filesystem\Exception\RuntimeException If unable to create the file.
+ * @throws Filesystem\Exception\RuntimeException If unable to create the file.
+ * @throws Filesystem\Exception\NotFoundException If $directory is not found.
+ * @throws Filesystem\Exception\NotDirectoryException If $directory is not a directory.
+ * @throws Filesystem\Exception\InvalidArgumentException If $prefix contains a directory separator.
  * @throws Psl\Exception\InvariantViolationException If $directory doesn't exist or is not writable.
  * @throws Psl\Exception\InvariantViolationException If $prefix contains a directory separator.
```

Comment:
I think these cases are already covered by `NotFoundException` and `InvalidArgumentException`, except for “is not writable” part.

Reply:
working on a PR to rework exceptions for File component as well, this is just done to satisfy psalm :)